docs/source/reference/debugging.rst

```diff
@@ -0,0 +1,200 @@
+**********************
+Debugging The Compiler
+**********************
+
+Performance
+===========
+
+The compiler has the ``--timing`` flag to dump timing information collected during operation.
+
+The output documents, in reverse chronological order, the cumulative time taken for the operation (and sub operations) to complete.
+Sub levels are indicated by successive repetitions of ``+``.
+
+
+Logging
+=======
+
+The compiler logs various categories of information during operation at various levels.
+
+Log levels are characterised by two things:
+
++ a dot-separated path of ever finer topics of interest e.g. scope.let
++ a natural number corresponding to the verbosity level e.g. 5
+
+If the user asks for some logs by writing::
+
+    %log scope 5
+
+they will get all of the logs whose path starts with `scope` and whose
+verbosity level is less or equal to `5`. By combining different logging
+directives, users can request information about everything (with a low
+level of details) and at the same time focus on a particular subsystem
+they want to get a lot of information about. For instance:::
+
+    %log 1
+    %log scope.let 10
+
+will deliver basic information about the various phases the compiler goes
+through and deliver a lot of information about scope-checking let binders.
+
+
+You can set the logging level at the command line using::
+
+    --log <level>
+
+and through the REPL using::
+
+    :log <string category> <level>
+
+    :logging <string category> <level>
+
+The supported logging categories are:
+
+.. csv-table:: Logging Categories
+   :header: "topic", "description"
+   :widths: 20, 20
+
+    ``auto``, some documentation of this option
```

Comment:
I'd rather not have this table duplicated in many different places.
It'd be better to have `idris2 --help logging` return the list of the
topics currently supported.

Reply:
#1467 implements `--help logging`

Reply:
great thanks!